edgelet/edgelet-docker/src/migc_persistence.rs

```diff
@@ -0,0 +1,312 @@
+use std::io::Write;
+use std::sync::{Arc, Mutex};
+use std::time::UNIX_EPOCH;
+use std::{collections::HashMap, fs, time::Duration};
+
+use edgelet_core::Module;
+use edgelet_settings::base::image::MIGCSettings;
+
+use crate::{DockerModule, Error};
+
+const TEMP_FILE: &str = "/tmp/images";
+
+#[derive(Debug, Clone)]
+struct MIGCPersistenceInner {
+    filename: String,
+    settings: Option<MIGCSettings>,
+}
+
+#[derive(Debug, Clone)]
+pub struct MIGCPersistence {
+    inner: Arc<Mutex<MIGCPersistenceInner>>,
+}
+
+impl MIGCPersistence {
+    pub fn new(filename: String, settings: Option<MIGCSettings>) -> Self {
+        Self {
+            inner: Arc::new(Mutex::new(MIGCPersistenceInner { filename, settings })),
+        }
+    }
+
+    pub async fn record_image_use_timestamp(
+        &self,
+        name_or_id: &str,
+        is_image_id: bool,
+        image_name_to_id: HashMap<String, String>, // HashMap<image_name, image_id>
+    ) {
+        if is_image_id {
+            self.write_image_use_to_file(name_or_id).await;
+        } else {
+            match image_name_to_id.get(name_or_id) {
+                Some(id) => {
+                    return self.write_image_use_to_file(id).await;
+                }
+                None => {
+                    log::error!("Could not find image with id: {}", name_or_id);
+                }
+            };
+        }
+    }
```

Comment:
If caller has available:
1. Whether it is an image_id
2. all image name -> id mappings
3. image name or id

Can't we simplify this by making caller just pass in an id?